share/patch/patch_db_zonemaster_backend_ver_8.2.0.pl

```diff
@@ -105,7 +105,7 @@ sub patch_db_postgresql {
         $dbh->do(
             q[
                 UPDATE test_results
-                SET domain = RTRIM('.', domain)
+                SET domain = RTRIM(domain, '.')
```

Comment:
RTRIM is not fully correct here. "iis.se..", "iis.se." and "iis.se" will all be stored as "iis.se", which is fine for "iis.se." because it was tested as "iis.se", but "iis.se.." was not.

Reply:
I guess the where clause could have been more restrictive. Do you know if we've had releases of Zonemaster that allowed zone names with double dots to be entered into the database?

Reply:
Good point. We could use for PostgreSQL `LEFT(domain, -1)` that removes the last character from the string.

Reply:
The problem is also present for MySQL, the LEFT() function does not work the same, but we could use something like `LEFT( domain, LENGTH(domain) - 1 )`.

Documentations for reference:
https://dev.mysql.com/doc/refman/5.7/en/string-functions.html
https://www.postgresql.org/docs/10/functions-string.html

Reply:
see #1031

Reply:
> I guess the where clause could have been more restrictive. Do you know if we've had releases of Zonemaster that allowed zone names with double dots to be entered into the database?

I am positive that Backend before some version ago let double dots through. One test could be if there are such entries in the zonemaster.net database. Please note that I have approved this PR anyway.